core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -211,6 +211,32 @@ private void cacheChanges() {
     this.cachedDeletes = deletes.build();
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o instanceof BaseSnapshot) {
+      BaseSnapshot other = (BaseSnapshot) o;
+      return this.snapshotId == other.snapshotId() &&
```

Comment:
Shouldn't this also include the the table  that was loaded? Or is it enough that id and parent id are almost certainly unique?

Reply:
I had the same thought. For the Beam use-case, this is sufficient. If you want to split out a single stream into multiple tables, then you would do a key-by on the destination table. The comparisons will be made within the table, therefore the sequenceNumber will make this unique. I'm happy to also add the table that was loaded, but this isn't in the class right now.

Reply:
I think it's okay for now, but it is an open question what makes two snapshots equal. Being careful here and including more fields as you do is pretty reasonable.